### ISSUE TITLE
Preserving Scroll Position Across Navigation in Vue.js with ViteSSG

### ISSUE THREAD

**Issue by WRXinYue** [YunYouJun/valaxy#334]
### Issue Description

I am encountering some challenges while using Vue.js and ViteSSG to build my project. I aim to achieve a functionality where, upon navigating from the homepage to an article page and then back to the homepage, the page remembers and scrolls to the original position. However, this functionality does not seem to be working currently.

#### Test Code

Here is the test code I used to implement this feature:

```vue
// layouts/home.vue
<script setup>
import { createRouter, onBeforeRouteLeave, onBeforeRouteUpdate, useRoute } from 'vue-router'
import { nextTick, onActivated, onMounted } from 'vue'
import { useSakuraAppStore } from '../stores/app'

const route = useRoute()
const sakura = useSakuraAppStore()
onMounted(async () => {
  const savedPosition = sakura.getScrollPosition(route.path)
  await nextTick()
  document.documentElement.scrollTop = savedPosition
})

onBeforeRouteLeave(() => {
  const currentScroll = document.documentElement.scrollTop
  console.log(currentScroll)
  sakura.setScrollPosition(route.path, currentScroll)
})
</script>
```

#### Identified Issue

I have noticed the following code in the Valaxy source, which might be the reason why my configuration is not effective:

```javascript
// not filter hide for ssg
const routesWithLayout = setupLayouts(import.meta.env.DEV
  ? routes
  : filterDraft(routes),
)

// https://github.com/antfu/vite-ssg
export const createApp = ViteSSG(
  App,
  {
    routes: routesWithLayout,
    base: import.meta.env.BASE_URL,
    scrollBehavior(to, from) {
      if (to.path !== from.path)
        return { top: 0 }
    },
  },
  (ctx) => {
    // app-level provide
    const { app } = ctx
    const config = initValaxyConfig()
    app.provide(valaxyConfigSymbol, config)

    registerComponents(ctx)
    setupMain(ctx, config)
  },
)
```

#### Request for Assistance

I am unsure about how to modify the ViteSSG configuration, or register a global route guard in the Valaxy theme, to achieve the desired functionality. Any suggestions?

**Comment by YunYouJun** [comment]
https://github.com/YunYouJun/valaxy/blob/1291dc4cef73dbf0977f2dec3ae4a09a0b6afda3/packages/valaxy-theme-press/setup/main.ts#L25

Is it what you want？

**Comment by WRXinYue** [comment]
Thanks a million!!!

**Comment by YunYouJun** [comment]
Where do you think I should add it to the document to make it easier for you to find?

**Comment by WRXinYue** [comment]
When I was browsing through the documentation, I indeed found information about the configuration of this extension. The file `setup/main.ts` is also present in `valaxy-theme-yun`. At that time, I guessed that this file could be used to configure the router, but I didn't know how to use it. Your recent explanation has perfectly solved the problem of me lacking examples. Furthermore, the page [https://valaxy.site/guide/custom/extend](https://valaxy.site/guide/custom/extend) also provides detailed information about router-related settings. I believe it was my oversight.
